### ISSUE TITLE
If not #3, then replace ResultValue<R> with a boolean signal

### ISSUE THREAD

**Issue by fabiancook** [opennetwork/dataset#4]
Use

```ts
export type ResultValue<Async extends boolean, V, PV = Promise<V>> = Async extends true ? V : PV;
```

Rather than

```ts
export type ResultType = Promise<any> | undefined;
export type ResultValue<R extends ResultType, V, PV = Promise<V>> = R extends undefined ? V : PV;
```

**Comment by fabiancook** [comment]
This reduces change on the codebase significantly, makes it a little easier to understand

```ts
export type ResultType = true | false;
export type ResultValue<R extends ResultType, V, PV = Promise<V>> = R extends true ? PV : V;
```

**Comment by fabiancook** [comment]
Closed by 3367d2425fe1
